### PR TITLE
Idempotent supplementary evidence with OCR updates

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-951

### Change description ###

comparing case document control_number numbers with exception

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 1]
```diff
@@ -420,30 +421,55 @@ private void updateSupplementaryEvidence(
         CaseDetails exceptionRecordDetails,
         boolean ignoreWarnings
     ) {
-        Optional<String> attachedToCase = getCaseExceptionRecordIsAttachedTo(
-            callBackEvent.exceptionRecordJurisdiction,
-            callBackEvent.exceptionRecordId
+        CaseDetails fetchedExceptionRecord = ccdApi.getCase(
+            callBackEvent.exceptionRecordId.toString(),
+            callBackEvent.exceptionRecordJurisdiction
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: not sure this is correct. You are fetching exception record but that is already passed in the callback ?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: trying to get existing case

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java [hunk 2]
```diff
@@ -427,13 +428,36 @@ private void updateSupplementaryEvidence(
 
         if (attachedToCase.isPresent()) {
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java]: of this is present it means it is updated and status already changed.
if it is present we should trow an exception.

**Comment by mdayican** [comment]
will create new one.

**Pull request closed without merging**